pkg/controller/controller_instance.go

```diff
@@ -644,30 +628,97 @@ func (c *controller) reconcileServiceInstance(instance *v1alpha1.ServiceInstance
 
 			return err
 		}
-		request.OriginatingIdentity = originatingIdentity
+	}
+
+	var (
+		isProvisioning             bool
+		provisionRequest           *osb.ProvisionRequest
+		updateRequest              *osb.UpdateInstanceRequest
+		currentOperation           v1alpha1.ServiceInstanceOperation
+		provisionOrUpdateText      string
+		provisionedOrUpdatedText   string
+		provisioningOrUpdatingText string
+	)
+	if toUpdate.Status.ReconciledGeneration == 0 {
+		isProvisioning = true
+		// osb client handles whether or not to really send this based
+		// on the version of the client.
+		requestContext := map[string]interface{}{
+			"platform":  ContextProfilePlatformKubernetes,
+			"namespace": instance.Namespace,
+		}
+		provisionRequest = &osb.ProvisionRequest{
+			AcceptsIncomplete:   true,
+			InstanceID:          instance.Spec.ExternalID,
+			ServiceID:           serviceClass.Spec.ExternalID,
+			PlanID:              servicePlan.Spec.ExternalID,
+			Parameters:          parameters,
+			OrganizationGUID:    string(ns.UID),
+			SpaceGUID:           string(ns.UID),
+			Context:             requestContext,
+			OriginatingIdentity: originatingIdentity,
+		}
+		currentOperation = v1alpha1.ServiceInstanceOperationProvision
+		provisionOrUpdateText = "provision"
+		provisionedOrUpdatedText = "provisioned"
+		provisioningOrUpdatingText = "provisioning"
+	} else {
+		isProvisioning = false
+		updateRequest = &osb.UpdateInstanceRequest{
+			AcceptsIncomplete:   true,
+			InstanceID:          instance.Spec.ExternalID,
+			ServiceID:           serviceClass.Spec.ExternalID,
+			OriginatingIdentity: originatingIdentity,
+		}
+		// Only send the plan ID if the plan name has changed from what the Broker has
+		if toUpdate.Status.ExternalProperties == nil ||
+			toUpdate.Status.InProgressProperties.ExternalServicePlanName != toUpdate.Status.ExternalProperties.ExternalServicePlanName {
+			planID := servicePlan.Spec.ExternalID
```

Comment:
We need to check the flag `plan_updatable` in `ServiceClass.Spec.PlanUpdatable`

Reply:
I don't think that this is the right place to check this. The change to the plan name has already been accepted by the API Server, given the state of the resources at that time. I don't think it is appropriate to ignore the accepted change to the plan name given the state of the resources at this time. We would need to do more than just not send the plan change to the Broker if that were the case. We would also need to somehow revert the plan change in the Spec of the ServiceInstance.